src/flowMax.coffee

```diff
@@ -1,31 +1,48 @@
 import {isAddPropTypes} from './addPropTypes'
 import {isRenderNothing} from './renderNothing'
 import {isReturns} from './returns'
+import {isAddWrapper} from './addWrapper'
+
+getArgumentsPropertyName = '__ad-hok-flowMax-getArguments'
+
+isFlowMax = (func) ->
+  func[getArgumentsPropertyName]
 
 flowMax = (...funcs) ->
-  flowLength = funcs?.length ? 0
-  if flowLength
-    for func, funcIndex in funcs when isAddPropTypes(func)
-      precedingFuncs =
-        if funcIndex is 0
-          []
-        else
-          funcs[0...funcIndex]
-      return flowMax ...precedingFuncs, func flowMax ...funcs[(funcIndex + 1)..]
-  (...args) ->
+  getPrecedingFuncs = (index) ->
+    if index is 0
+      []
+    else
+      funcs[0...index]
+  getFollowingFuncs = (index) ->
+    funcs[(index + 1)..]
+  ret = (...args) ->
+    return args[0] unless funcs?.length
+    flowLength = funcs.length
     index = 0
-    props =
-      if flowLength
-        funcs[0] ...args
-      else
-        args[0]
-    return null if isRenderNothing props
-    return returnsVal[1] if (returnsVal = isReturns props)
-    while ++index < flowLength
+    props = null
+    while index < flowLength
       func ###:### = funcs[index]
-      props = func props
+      if (getNestedFlowMaxArguments = isFlowMax func)
+        funcs ### : ### = [
+          ...getPrecedingFuncs(index)
+          ...getNestedFlowMaxArguments()
```

Comment:
Nested `flowMax()`s get inlined "persistently" on the first invocation

This should work for arbitrarily nested `flowMax()`'s?